src/main/kotlin/commands/Funcommands.kt

```diff
@@ -0,0 +1,100 @@
+package chattore.commands
+
+import chattore.ChattORE
+import chattore.FunCommandConfig
+import chattore.ChattoreException
+import com.uchuhimo.konf.Config
+import com.velocitypowered.api.proxy.Player
+import co.aikar.commands.BaseCommand
+import co.aikar.commands.annotation.*
+import chattore.entity.ChattORESpec
+import chattore.toComponent
+import chattore.render
+import net.kyori.adventure.text.Component
+import net.kyori.adventure.text.event.ClickEvent
+import net.kyori.adventure.text.event.HoverEvent
+
+@CommandAlias("funcommands|fc")
+@CommandPermission("chattore.funcommands")
+class Funcommands(
+    private val config: Config,
+    private val chattORE: ChattORE,
+    private val commands: List<FunCommandConfig>
+) : BaseCommand() {
+
+    @Default
+    @Description("Displays information about the /funcommands command")
+    fun onDefault(player: Player) {
+        val message = config[ChattORESpec.format.funcommandsDefault].render(
+            mapOf(
+                "sender" to player.username.toComponent()
+            )
+        )
+        player.sendMessage(message)
+
+    }
+
+    @Subcommand("list")
+    @Description("Lists all available fun commands in reverse alphabetical order")
+    fun onList(player: Player) {
+        if (commands.isEmpty()) {
+            val noCommandsMessage = config[ChattORESpec.format.funcommandsNoCommands].render(
+                mapOf(
+                    "sender" to player.username.toComponent()
+                )
+            )
+            player.sendMessage(noCommandsMessage)
+            return
+        }
+
+        val headerMessage = config[ChattORESpec.format.funcommandsHeader].render(
+            mapOf(
+                "sender" to player.username.toComponent()
+            )
+        )
+        player.sendMessage(headerMessage)
+
+        val commandsMessage = Component.text()
+        commands.sortedByDescending { it.command }.forEach { cmd ->
+            val clickableCommand = Component.text("/${cmd.command} ")
+                .hoverEvent(
+                    HoverEvent.showText(
+                        Component.text(cmd.description)
+                    )
+                )
+                .clickEvent(
```

Comment:
This should instead be a [suggest action](https://jd.advntr.dev/api/latest/net/kyori/adventure/text/event/ClickEvent.Action.html#SUGGEST_COMMAND). It may also have to fill out arg1 and subsequent args if necessary, depending on the command.